### PR TITLE
Use Spago instead of psc-package

### PR DESCRIPTION
**Pull request by mhmdanas**
~~This is my work on migrating to Spago so far, but I'm keeping it as a draft so I can polish it later.~~

Benefits of migrating to Spago:
- We don't have to use both Pulp and psc-package
- It's far easier to customize the package set with Spago than with psc-package

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,13 +23,13 @@
   ],
   "scripts": {
     "setup": "npm run build",
-    "start": "npm run setup && run-p server esbuild:watch",
-    "esbuild:watch": "pulp -w --then 'npm run esbuild' build -m Insect",
+    "start": "node copy && run-p server esbuild:watch",
+    "esbuild:watch": "spago build -w --then 'npm run esbuild'",
     "esbuild": "esbuild output/Insect/index.js --bundle --minify --outfile=web/insect.js --target=es6 --global-name=Insect",
```
Comment (mhmdanas) [inline package.json:28]: I could've used `spago bundle-app` here, but I like that we can pass whatever arguments to `esbuild` we want (I think we cannot do that with `spago bundle-app`).

File: .purs-repl [hunk 2]
```diff
@@ -0,0 +1 @@
+import Prelude
```
Comment (sharkdp) [inline .purs-repl:1]: can you explain what this does? Is it like a `.bashrc` file for the PureScript REPL? If so, … is the Prelude not included by default there?

File: packages.dhall [hunk 3]
```diff
@@ -1,16 +1,105 @@
...
+          , "datetime"
+          , "now"
+          ]
+      , repo =
+          "https://github.com/hdgarrood/purescript-benchotron.git"
+      , version =
+          "v7.0.0"
+      }
+-------------------------------
+-}
```
Comment (sharkdp) [inline packages.dhall:100]: maybe remove this?

File: spago.dhall [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+You can edit this file as you like.
+
+Need help? See the following resources:
+- Spago documentation: https://github.com/purescript/spago
+- Dhall language tour: https://docs.dhall-lang.org/tutorials/Language-Tour.html
+
+When creating a new Spago project, you can use
+`spago init --no-comments` or `spago init -C`
+to generate this file without the comments in this block.
+-}
```
Comment (sharkdp) [inline spago.dhall:12]: Similar here?

File: .purs-repl (hunk 2 above)
Comment (sharkdp) [reply .purs-repl:1]: If it doesn't cause too much pain, I would be in favor of removing this file. Keep the root folder clean.

Comment (mhmdanas) [reply .purs-repl:1]: > can you explain what this does? Is it like a .bashrc file for the PureScript REPL? If so, … is the Prelude not included by default there?

Kind of, yes. And no, `Prelude` isn't imported by default, unless you're running `spago repl` outside of a Spago project.

> If it doesn't cause too much pain, I would be in favor of removing this file. Keep the root folder clean.

I wouldn't mind removing it too much. I rarely use the REPL, at least with Insect.

**Comment by mhmdanas** [comment]
I resolved your review comments. Anything else you'd like to ask about or suggest?

**Review by sharkdp — APPROVED** [review]
No, this looks good to me! Feel free to merge.

**Pull request merged**